java/acorn-core/src/main/java/com/datasqrl/ai/models/ChatProvider.java

```diff
@@ -10,25 +10,10 @@
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 
-@AllArgsConstructor
-public abstract class ChatProvider<Message, FunctionCall> {
+public interface ChatProvider {
 
-  public static int DEFAULT_HISTORY_LIMIT = 50;
-  public static final int FUNCTION_CALL_RETRIES_LIMIT = 5;
+  GenericChatMessage chat(String message, Context context);
```

Comment:
@mbroecheler Should we replace Map<String, Object> with Context everywhere now? I saw you didn't follow through down to the models, i.e. `GenericChatMessage`. Is that on purpose?

Reply:
Good observation. I reverted the change in GenericChatMessage because it become difficult to de/serialize and we don't read the context back in (for security reasons), so the value isn't there.